src/app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks.ts

```diff
@@ -60,6 +61,21 @@ function useNativeSegwitSigner(accountIndex: number) {
   }, [accountIndex, accountKeychain, network.chain.bitcoin.network]);
 }
 
+export function useCurrentAccountNativeSegwitDetails() {
+  const currentAccountIndex = useCurrentAccountIndex();
+  const currentNetworkConfiguration = useCurrentNetwork();
+  const currentBitcoinNetwork = currentNetworkConfiguration.chain.bitcoin.network;
+  const currentKeychain =
+    useNativeSegwitActiveNetworkAccountPrivateKeychain()?.(currentAccountIndex);
+  if (!currentKeychain) return;
+  const currentAddressIndexKeychain =
+    deriveAddressIndexKeychainFromAccount(currentKeychain)(currentAccountIndex);
+  const currentPayment = getNativeSegWitPaymentFromAddressIndex(currentAddressIndexKeychain, currentBitcoinNetwork);
+  const currentAddress = currentPayment.address;
+
+  return { currentBitcoinNetwork, currentAddress, currentAddressIndexKeychain };
+}
```

Comment:
Ah, yeah you need the private key. Maybe you can pull this directly from `useNativeSegwitActiveNetworkAccountPrivateKeychain()`, but the other values could be pulled from the signer, to avoid more address derivation logic?